cabal.project.freeze

```diff
@@ -0,0 +1,84 @@
+active-repositories: hackage.haskell.org:merge
```

Comment:
To be clear, the reason I included it is just because "it's what people expect." When people use Flakes, there's a lockfile that exactly specifies dependencies are needed; so similarly I created a cabal project lockfile ("freeze file") for non-Nix users. That's really the only reason; so that both methods have "a fixed snapshot of dependencies that they need" (even if they aren't perfectly aligned.)